packages/pds/tests/account-deletion.test.ts

```diff
@@ -0,0 +1,228 @@
+        .execute(),
+      db.db
+        .selectFrom('ipld_block')
+        .where('creator', '=', carol.did)
+        .selectAll()
+        .execute(),
+    ])
+
+    expect(roots.length).toBe(0)
+    expect(users.length).toBe(0)
+    expect(blocks.length).toBe(0)
+  })
+
+  it('no longer stores indexed records from the user', async () => {
+    const [posts, votes, reposts, follows] = await Promise.all([
+      db.db
+        .selectFrom('post')
+        .where('creator', '=', carol.did)
+        .selectAll()
+        .execute(),
+      db.db
+        .selectFrom('vote')
+        .where('creator', '=', carol.did)
+        .selectAll()
+        .execute(),
+      db.db
+        .selectFrom('repost')
+        .where('creator', '=', carol.did)
+        .selectAll()
+        .execute(),
+      db.db
+        .selectFrom('follow')
+        .where('creator', '=', carol.did)
+        .selectAll()
+        .execute(),
+
+      db.db
+        .selectFrom('user')
+        .where('handle', '=', carol.handle)
+        .selectAll()
+        .execute(),
+      db.db
+        .selectFrom('ipld_block')
+        .where('creator', '=', carol.did)
+        .selectAll()
+        .execute(),
+    ])
+
+    expect(posts.length).toBe(0)
+    expect(votes.length).toBe(0)
+    expect(reposts.length).toBe(0)
+    expect(follows.length).toBe(0)
+  })
+
+  it('deletes relevant blobs', async () => {
+    const imgs = sc.posts[carol.did][0].images
+    // carols first blob is used by other accounts
+    const first = CID.parse(imgs[0].image.cid)
+    // carols second blob is used by only her
+    const second = CID.parse(imgs[1].image.cid)
+    const got = await blobstore.getBytes(first)
+    expect(got).toBeDefined()
+    const attempt = blobstore.getBytes(second)
+    await expect(attempt).rejects.toThrow(BlobNotFoundError)
+
+    const [repoBlobs, blobs] = await Promise.all([
+      db.db
+        .selectFrom('repo_blob')
+        .where('did', '=', carol.did)
+        .selectAll()
+        .execute(),
+      db.db
+        .selectFrom('blob')
+        .where('cid', 'in', [first.toString(), second.toString()])
+        .selectAll()
+        .execute(),
+    ])
+
+    expect(repoBlobs.length).toBe(0)
+    expect(blobs.length).toBe(1)
+    expect(blobs[0].cid).toEqual(first.toString())
+  })
+
+  it('no longer displays the users posts in feeds', async () => {
+    const feed = await client.app.bsky.feed.getTimeline(undefined, {
+      headers: sc.getHeaders(sc.dids.alice),
+    })
+    const found = feed.data.feed.filter(
+      (item) => item.post.author.did === carol.did,
+    )
+    expect(found.length).toBe(0)
+  })
+
+  it('removes notifications from the user', async () => {
+    const notifs = await client.app.bsky.notification.list(undefined, {
+      headers: sc.getHeaders(sc.dids.alice),
+    })
+    const found = notifs.data.notifications.filter(
+      (item) => item.author.did === sc.dids.carol,
+    )
+    expect(found.length).toBe(0)
+  })
+})
```

Comment:
Only thing I might consider adding to these tests is confirmation that other users' data is still around.  Touched on it a little bit in the blob test.  It might look like confirming that exactly Alice's blocks are missing from the db.

Reply:
yeah good looks

I changed my approach to it a bit & also added tests for more tables as well. Fortunately didn't come across any additional bugs from it 😛 

https://github.com/bluesky-social/atproto/pull/488/commits/d409acbff1e29ecb92d7bfa6de358cc38641c793